### PR TITLE
Dockerfile-lint-3-2

### PR DESCRIPTION
**Pull request by ammarlakis**
**Description**

Changes proposed in this pull request:

- Fixing linting errors in Dockerfiles.

**Related issue(s)**

Split from #12964 
Related to #12854

### CODE REVIEW

File: tools/gitserver/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
 FROM alpine:3.12
 
-RUN apk update &&\
-  apk add git-daemon apache2 apache2-utils
+FROM alpine:3.15
+
+RUN apk add --no-cache git-daemon apache2 apache2-utils && \
+    apk upgrade --no-cache --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community \
```
Comment (dbadura) [inline tools/gitserver/Dockerfile:6]: Do we need this?

Comment (ammarlakis) [reply tools/gitserver/Dockerfile:6]: yes, it has a critical vulnerability according to protecode scan so I had to update it to the latest, and the latest version is in edge repo.

Comment (dbadura) [reply tools/gitserver/Dockerfile:6]: Maybe I am blind, but I don't see that after adding repo `edge` we install something from this repo.
Also this image is used only in tests.

Comment (ammarlakis) [reply tools/gitserver/Dockerfile:6]: see next line, I'm upgrading `expat` library, which was already there (I guess a dependency of apache2) to the latest version.

**Pull request merged**